File: 2661-first-completely-painted-row-or-column/2661-first-completely-painted-row-or-column.java
class Solution {
    public int firstCompleteIndex(int[] arr, int[][] mat) {
        int m = mat.length, n = mat[0].length;
        Map<Integer, int[]> mp = new HashMap<>();
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                mp.put(mat[i][j], new int[]{i, j});
            }
        }
        int[] rowCount = new int[m];
        int[] colCount = new int[n];
        for (int i = 0; i < arr.length; i++) {
            int val = arr[i];
            int[] pos = mp.get(val);
            int row = pos[0], col = pos[1];
            rowCount[row]++;
            colCount[col]++;
            if (rowCount[row] == n || colCount[col] == m) {
                return i;
            }
        }
        return -1;
    }
}

File: LeetCode/2661-first-completely-painted-row-or-column/2661-first-completely-painted-row-or-column.py
class Solution:
    def firstCompleteIndex(self, arr: List[int], mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        mp = {mat[i][j]: (i, j) for i in range(m) for j in range(n)}
        rowCount = [0] * m
        colCount = [0] * n
        for i, val in enumerate(arr):
            row, col = mp[val]
            rowCount[row] += 1
            colCount[col] += 1
            if rowCount[row] == n or colCount[col] == m:
                return i
        return -1